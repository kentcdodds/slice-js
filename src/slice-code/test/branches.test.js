import {curry} from 'lodash'
import {snapSlice} from './helpers/utils'

const branchesFilename = require.resolve('./fixtures/branches')
const snapBranchesSlice = curry(snapSlice)(branchesFilename)

test(
  'ifOnly(true)',
  snapBranchesSlice(({ifOnly}) => ifOnly(true))
)

test(
  'ifOnly(false)',
  snapBranchesSlice(({ifOnly}) => ifOnly(false))
)

test(
  'ifElse(true)',
  snapBranchesSlice(({ifElse}) => ifElse(true))
)

test(
  'ifElse(false)',
  snapBranchesSlice(({ifElse}) => ifElse(false))
)

test(
  'ifElseIfElse(true, true)',
  snapBranchesSlice(({ifElseIfElse}) => ifElseIfElse(true, true))
)

test(
  'ifElseIfElse(true, false)',
  snapBranchesSlice(({ifElseIfElse}) => ifElseIfElse(true, false))
)

test(
  'ifElseIfElse(false, true)',
  snapBranchesSlice(({ifElseIfElse}) => ifElseIfElse(false, true))
)

test(
  'ifElseIfElse(false, false)',
  snapBranchesSlice(({ifElseIfElse}) => ifElseIfElse(false, false))
)
