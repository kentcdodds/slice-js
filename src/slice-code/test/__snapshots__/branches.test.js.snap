exports[`test ifElse(false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifElse(passIf) {
  return !passIf;
}"
`;

exports[`test ifElse(true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifElse(passIf) {
  return passIf;
}"
`;

exports[`test ifElseIfElse(false, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifElseIfElse(passIf1, passIf2) {
  return !passIf1 && !passIf2;
}"
`;

exports[`test ifElseIfElse(false, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifElseIfElse(passIf1, passIf2) {
  return passIf2;
}"
`;

exports[`test ifElseIfElse(true, false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifElseIfElse(passIf1, passIf2) {
  return passIf1;
}"
`;

exports[`test ifElseIfElse(true, true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifElseIfElse(passIf1, passIf2) {
  return passIf1;
}"
`;

exports[`test ifOnly(false) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifOnly(passIf) {}"
`;

exports[`test ifOnly(true) 1`] = `
"/* eslint no-constant-condition:0, consistent-return:0 */
export { ifOnly, ifElse, ifElseIfElse };

function ifOnly(passIf) {
  if (passIf) {
    return passIf;
  }
}"
`;
